### DISCUSSION TITLE
Change port to interface to the web server

### DISCUSSION THREAD

**Discussion by FrancescoCrevatin** [pschatzmann/arduino-audio-tools#461] [Q&A] [answered]
Hi, first of all thanks for the examples and the code you posted, they are very helpful.

I wanted to ask you: I'm trying this [example ](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-webserver/streams-i2s-webserver_wav/streams-i2s-webserver_wav.ino)that creates the web server with the ESP32 IP address and port 80, but if I wanted to change the port to interface with the web server, how could I do it? Which file should I go to?

Thank you for your availability and for all the work you have done!

**Comment by pschatzmann** [comment, marked as answer]
This is currently hard coded in the AudioServer class: the class variables have WiFiServer server = WiFiServer(80);
I convert this to an issue because I think this should be customizable

**Reply by FrancescoCrevatin** [reply]
> This is currently hard coded in the AudioServer class: the class variables have WiFiServer server = WiFiServer(80); I convert this to an issue because I think this should be customizable

Perfect, thanks. I'll see if it works as soon as I figure out how to upgrade the ESP32 GCC. Indeed, could you give me one last hand with this problem? I don't understand where to go and how to do this GCC update. I've already looked at the Espressif documentation and various Topics but I haven't been able to do anything. Could you tell me what should I do?

Thank you very much for your help and availability

**Reply by pschatzmann** [reply]
I don't understand: why would you want to upgrade GCC ? 
Are you working in Arduino ?

**Reply by FrancescoCrevatin** [reply]
> I don't understand: why would you want to upgrade GCC ? Are you working in Arduino ?

I need to upgrade my GCC because I'm having a problem running this code on my ESP32, which errors are multiple and are almost identical (only the variable names change): `'audio_tools::mime_pcm' declared as an 'inline' variable
 INLINE_VAR const char* mime_pcm = "audio/pcm";`.
At least from what I've seen on the internet I have to upgrade the GCC, but I really don't know what to do.

**Comment by pschatzmann** [comment]
What you see are warnings and not errors. You don't need to upgrade. You can to one of the following:
- Use the latest version of the AudioTools. I changed the default setting not to use inline variables any more
- Change the setting AudioConfig.h not to use inline variables yourself on your installed version
- In Arduino Settings set Compiler Warnings to normal and make sure that you have installed the latest ESP32 Version in the Board Manager

**Reply by FrancescoCrevatin** [reply]
Ah ok. Which is the latest version of the AudioTools?

**Reply by pschatzmann** [reply]
I haven't released these changes yet, so it would be the main Branch

**Reply by FrancescoCrevatin** [reply]
> I haven't released these changes yet, so it would be the main Branch

Ok, can you let me know when you release it? So, since I can't take stock one of your solutions, what parameter should I change in the AudioConfig.h?

Thanks again for your availability.

**Reply by pschatzmann** [reply]
what do you mean with stock one ?. You can download it with git or via the green Code button in Github.
The setting is from
```
#  define USE_INLINE_VARS 1
to 
#  define USE_INLINE_VARS 0
```

**Reply by FrancescoCrevatin** [reply]
> what do you mean with stock one ?. You can download it with git or via the green Code button in Github. The setting is from
> 
> ```
> #  define USE_INLINE_VARS 1
> to 
> #  define USE_INLINE_VARS 0
> ```

I meant: "since the first solution I can't use it because you still have to release the version, what parameters.."
Anyway thank you very much, now I no longer have any error messages and I have the stream on the WebServer, only that you hear almost nothing.
Do you have any tips on how to turn up the volume or how to improve the INMP44's capture of surrounding noise?

**Comment by FrancescoCrevatin** [comment]
Last thing: if I wanted to use DMA for greater efficiency, in which file should I make changes? Or do you already have an example using the Web server and DMA?

Thanks for all the help

**Comment by pschatzmann** [comment]
Are you working with an ESP32 ? This does not have any generic DMA API like e.g. the STM32 Microcontrollers. So I am not sure what you are referring to...

**Reply by FrancescoCrevatin** [reply]
Yes you are right, I was wrong.

Now how can I improve the sensitivity of the microphone? Use an amplifier? Change any parameters?
Then I tried port forwarding and it turns out that the mic is stuttering.. any idea how to fix this?

Thank you

**Comment by pschatzmann** [comment]
You would just turn up the volume on your computer. Alternaively you can try to modify the Volume as described in the [Wiki](https://github.com/pschatzmann/arduino-audio-tools/wiki/Volume-Control)

**Reply by FrancescoCrevatin** [reply]
Thank you for everything. You helped me a lot
